src/gt4py/next/iterator/ir_utils/domain_utils.py

```diff
@@ -144,28 +137,79 @@ def translate(
+                    new_range = SymbolicRange(
+                        im.literal(str(0), builtins.INTEGER_INDEX_BUILTIN),
+                        im.ensure_expr(symbolic_domain_sizes[new_dim.value]),
+                    )
+                else:
+                    assert common.is_offset_provider(offset_provider)
+                    connectivity = offset_provider[off.value]
+                    assert isinstance(connectivity, common.Connectivity)
+                    skip_value = connectivity.skip_value
+
+                    # fold & convert expr into actual integers
+                    range_exprs = new_ranges[old_dim].start, new_ranges[old_dim].stop
+                    range_exprs = tuple(
+                        collapse_tuple.CollapseTuple.apply(
+                            expr,
+                            within_stencil=False,
+                            allow_undeclared_symbols=True,
+                        )
+                        for expr in range_exprs
+                    )  # type: ignore[assignment]  # mypy not smart enough
+                    assert all(isinstance(expr, itir.Literal) for expr in range_exprs)
+                    start, stop = (int(literal.value) for literal in range_exprs)  # type: ignore[attr-defined]  # mypy does not understand assert above
+
+                    nb_index: slice | int
+                    if val in [trace_shifts.Sentinel.ALL_NEIGHBORS, trace_shifts.Sentinel.VALUE]:
+                        nb_index = slice(None)
+                    else:
+                        nb_index = val.value  # type: ignore[assignment]  # assert above
+
+                    accessed = connectivity.ndarray[start:stop, nb_index]
+
+                    if isinstance(val, itir.OffsetLiteral) and np.any(accessed == skip_value):
+                        # TODO(tehrengruber): Turn this into a configurable error. This is currently
+                        #  not possible since some test cases starting from ITIR containing
+                        #  `can_deref` might lead here. The frontend never emits such IR and domain
+                        #  inference runs after we transform reductions into stmts containing
+                        #  `can_deref`.
+                        warnings.warn(
+                            UserWarning(
+                                f"Translating '{self.as_expr()}' using '{shift[0].value}' has "
+                                f"an out-of-bounds access."
+                            ),
+                            stacklevel=2,
+                        )
+
+                    new_start, new_stop = accessed.min(), accessed.max() + 1  # type: ignore[attr-defined]  # TODO(havogt): improve typing for NDArrayObject
+
+                    fraction_accessed = np.unique(accessed).size / (new_stop - new_start)  # type: ignore[call-overload]  # TODO(havogt): improve typing for NDArrayObject
```

Comment:
```suggestion
                    fraction_accessed = xp.unique(accessed).size / (new_stop - new_start)  # type: ignore[call-overload]  # TODO(havogt): improve typing for NDArrayObject
```
?
or where do you make sure it lives on host?